report/3.Theory/LanguageSpecification.tex

```diff
@@ -126,6 +132,29 @@ \section{Language Specification}
 Not having operators for arithmetic operations also means that there are no native operator precedence in the language.
 This can be amended by structuring calculations in a left-to-right manner, or by encapsulating operations with parenthesis. 
 
+\textbf{High order functions}\\
+As mentioned in section \ref{subsec:hofhcl} the language will include high-order functions. 
+To declare that a function will accept other functions the developer should simply declare a parameter of type "func" with the expected input and output types from the function.
+
+Both predeclared functions and anonymous in-line functions can be passed as parameters. 
+However, if using a predeclared function, the identifier should be prefixed with a colon to indicate that the function should be passed and not invoked. 
+Since lambdas should not be prefixed with a colon, this makes it impossible for a lambda to be invoked in-line. 
+Therefore lambdas may only be passed as parameters, and never invoked by itself.
```

Comment:
the shouln't be comma before and

iguess since its lambdas in plural, it should be 'themselves'?